src/App.tsx

```diff
@@ -1,15 +1,51 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
 import './App.css'
 import UsersList from './features/users'
+import { AppDispatch, RootState } from './app/store/store'
+import { useDispatch } from 'react-redux'
+import { useSelector } from 'react-redux'
+import { useEffect } from 'react'
+import { getRestaurants } from './features/restaurants/restaurantsSlice'
 
 function App() {
   const dispatch = useDispatch<AppDispatch>()
+  const { status, restaurants, error } = useSelector((state: RootState) => state.restaurants)
 
+  useEffect(() => {
+    if (status === 'idle') {
+      dispatch(getRestaurants())
+    }
+  }, [status, dispatch])
```

Comment:
Would love to talk through why 'dispatch' is needed here at the end of the useEffect. That's something that I couldn't figure out during the tutorial.

Reply:
Yeah, I thought it was weird too! It's not for functionality but for linting errors. `ESLint` will say you have a missing dependency in your array. It assumes that any outside function or variable you use could change and needs to be tracked, even though we know `dispatch` won't change. We can test out and see if we get any errors! @Nagel29

Reply:
Ahh gotcha, thank you for explaining that! That totally makes sense now!